src/ps_endpointer.c

```diff
@@ -241,13 +249,19 @@ ps_endpointer_end_stream(ps_endpointer_t *ep,
                 ep->frame_size);
         return NULL;
     }
-    
+
     if (out_nsamp)
         *out_nsamp = 0;
     if (!ep->in_speech)
         return NULL;
     ep->in_speech = FALSE;
-    ep->speech_end = ep->qstart_time;
+
+    /* Use callback timestamp if available */
+    if (ep->timestamp_cb != NULL) {
+        ep->speech_end = ep->timestamp_cb(ep->timestamp_cb_data);
```

Comment:
Do we need to adjust for the time elapsed since this frame here, like we do below?

Reply:
Good catch!